### ISSUE TITLE
How to force loading the language file

### ISSUE THREAD

**Issue by twolff-iow** [angular-translate/angular-translate#1278]
Hi,

I have an application which makes heavy use of customizable components. My problem now is that the translation file is always loaded a bit too late and some controllers of these components are already executed and their request to get translations via $translate.instant just returns the translation id.

In my .config function of the application I configure the $translateProvider as this

```
            $translateProvider
                .useStaticFilesLoader({prefix: 'i18n/', suffix: '.json'})
                .useSanitizeValueStrategy(null)
                .registerAvailableLanguageKeys(['en', 'ru'])
                .determinePreferredLanguage()
                .fallbackLanguage('en');
```

The workflow for executing the config/run methods and the controllers vs. loading the translation file with the static files loader goes as follows:

```
module1.module.config
module2.module.config
main_app.config
module1.module.run
module2.module.run
main_app.run
module1.controller ***
file.loader.success
module2.controller
```

So, the main application module resolves the dependencies and configures module1 and module2 and then configures itself (with the $translateProvider as stated above). Then module1 and module2 are run and afterwards main_app is run. Then controller of module1 is executed (***) and only afterwards the translation file is loaded. Hence, all translations I try to do with $translate.instant in the controller of module1 do not have any translations yet.

Can I somehow force the loading of the translation file to happen earlier?

Thanks,
   Tobias.

**Comment by knalli** [comment]
> My problem now is that the translation file is always loaded a bit too late and some controllers of these components are already executed and their request to get translations via $translate.instant just returns the translation id.

That is why you should not use `.instant()` without paying attribution with its implications. You should not ignore the warnings in the documenting and guide.

> Can I somehow force the loading of the translation file to happen earlier?

No, there is no such option. The languages are being tried to load as soon as possible. But we can neither pause the AngularJS config/run phases, nor we have actually the network under our control.

tl;dr: You have to deal that languages will eventually never be available at run phase, but later. This is asynchronous processing like in **A**JAX. In Angular (in this context), a run phase includes the initial constructors of controllers, services and directives as well.

In special cases like mobile hybrid apps you can preload the files and inject them statically (`$translateProvider.translation(lang, texts)`) and with a deferred AngularJS bootstrap (please have a look in the official AJS documentation).
